subsys/event_manager/event_manager_priv.h

```diff
@@ -16,6 +16,11 @@
 extern "C" {
 #endif
 
+#if !CONFIG_X86
+#define _EM_FORCED_ALIGNMENT
+#else
+#define _EM_FORCED_ALIGNMENT __aligned(4)
```

Comment:
Isn't there somewhere in Zephyr where we can take this from? Can you search to see if we have generic minimum alignment macros somewhere?

Reply:
> Isn't there somewhere in Zephyr where we can take this from?

Hi @carlescufi , as far as I can tell there is no other way to handle this.
The problem happens because of default compiler settings (`-malign-data` to be precise going to `compat` which goes to `cacheline` https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html). The problem is gone when I force it to `abi`.
The result is that whatever structure goes beyond 32 bytes will be automagically aligned to 64 bytes WITHOUT use knowing about this thing. The size and alignment of the data as seen from macros will be according to ABI. Zephyr has no way of knowing about this (except you know what x86 does to you structures).

Since we have no way of telling what is this rule (in generic case) we cannot make traverse code to jump to correct addresses using any generic macro, sizeof or alignof.

The other way around is to align the structures to some known alignment. This is what we did. It can be any alignment that would be smaller then expected structure size (4 is great). We could make it aligned to `sizeof(int)` or any random macro specifying the alignment but I don't think it will have much sense as there is no macro associated with this "feature" :(
So in my opinion 4 is as good value as any other.

Reply:
There is however another way of solving this and it is to specify `CONFIG_COMPILER_OPT="-malign-data=abi"` or even better make it added always whenever x86 is being used as a platform.

Reply:
Sooner or later somebody else may have similar problem with this thing.

What are your opinions on adding `-malign-data=abi` for x86 compilation? @carlescufi , @tejlmand